xen/runtime/ocaml/minor_gc.c

```diff
@@ -1,344 +0,0 @@
-   oldifying its fields. */
-void caml_oldify_mopup (void)
-{
-  value v, new_v, f;
-  mlsize_t i;
-
-  while (oldify_todo_list != 0){
-    v = oldify_todo_list;                /* Get the head. */
-    Assert (Hd_val (v) == 0);            /* It must be forwarded. */
-    new_v = Field (v, 0);                /* Follow forward pointer. */
-    oldify_todo_list = Field (new_v, 1); /* Remove from list. */
-
-    f = Field (new_v, 0);
-    if (Is_block (f) && Is_young (f)){
-      caml_oldify_one (f, &Field (new_v, 0));
-    }
-    for (i = 1; i < Wosize_val (new_v); i++){
-      f = Field (v, i);
-      if (Is_block (f) && Is_young (f)){
-        caml_oldify_one (f, &Field (new_v, i));
-      }else{
-        Field (new_v, i) = f;
-      }
-    }
-  }
-}
-
-/* Make sure the minor heap is empty by performing a minor collection
-   if needed.
-*/
-void caml_empty_minor_heap (void)
-{
-  value **r;
-
-  if (caml_young_ptr != caml_young_end){
-    caml_in_minor_collection = 1;
-    caml_gc_message (0x02, "<", 0);
-    caml_oldify_local_roots();
-    for (r = caml_ref_table.base; r < caml_ref_table.ptr; r++){
-      caml_oldify_one (**r, *r);
-    }
-    caml_oldify_mopup ();
-    for (r = caml_weak_ref_table.base; r < caml_weak_ref_table.ptr; r++){
-      if (Is_block (**r) && Is_young (**r)){
-        if (Hd_val (**r) == 0){
-          **r = Field (**r, 0);
-        }else{
-          **r = caml_weak_none;
-        }
-      }
-    }
-    if (caml_young_ptr < caml_young_start) caml_young_ptr = caml_young_start;
-    caml_stat_minor_words += Wsize_bsize (caml_young_end - caml_young_ptr);
-    caml_young_ptr = caml_young_end;
-    caml_young_limit = caml_young_start;
-    clear_table (&caml_ref_table);
-    clear_table (&caml_weak_ref_table);
-    caml_gc_message (0x02, ">", 0);
-    caml_in_minor_collection = 0;
-  }
-  caml_final_empty_young ();
-#ifdef DEBUG
-  {
-    value *p;
-    for (p = (value *) caml_young_start; p < (value *) caml_young_end; ++p){
-      *p = Debug_free_minor;
-    }
-    ++ minor_gc_counter;
-  }
-#endif
-}
-
-/* Do a minor collection and a slice of major collection, call finalisation
-   functions, etc.
-   Leave the minor heap empty.
-*/
-CAMLexport void caml_minor_collection (void)
-{
-  value *note_gc;
-  uint64_t start_time;
-
-  note_gc = caml_named_value("MProf.Trace.note_gc");
-  if (note_gc)
-    start_time = NOW();
-
-  intnat prev_alloc_words = caml_allocated_words;
-
-  caml_empty_minor_heap ();
-
-  caml_stat_promoted_words += caml_allocated_words - prev_alloc_words;
-  ++ caml_stat_minor_collections;
-  caml_major_collection_slice (0);
-  caml_force_major_slice = 0;
-
-  caml_final_do_calls ();
-
-  if (note_gc){
-    double duration_ns = (double) (NOW () - start_time);
-    value result = caml_callback_exn(*note_gc, caml_copy_double(duration_ns / 1000000000));
-    if (Is_exception_result(result))
-      printk("warning: note_gc threw an exception!\n");
-  }
```

Comment:
We are indeed, but we can introduce a local patch into the xen-ocaml repository that restores them before merging this.  Would you have a patch against vanilla 4.01.0 in a Gist that I could integrate?